rules/enforce-return-statement.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @fileoverview Enforce the use of a return statement in a function
+ * @author Nicolas Feignon <nfeignon@gmail.com>
+ */
+
+'use strict';
+
+module.exports = {
+
+    meta: {
+
+        docs: {
+            recommended: true,
+            type: 'error',
+            description: 'Enforce the use of a return statement in a function',
+        },
+
+        schema: []
+
+    },
+
+    create: function (context) {
+
+        function inspectFunctionDeclaration(emitted) {
+            var node = emitted.node;
+            var found = false;
+
+            if (!emitted.exit) {
+                for (let expr of node.body.body) {
+                    if (expr['type'] === 'ReturnStatement')
+                        found = true;
+                }
+
+                if (!found) {
+                    context.report({
+                        node: node,
+                        message: 'Missing return statement in function'
```

Comment:
`Missing return statement in function <FUNCTION NAME>` (OR `Missing return statement in fallback function` in the special case of fallback function, which doesn't have a name)